File: examples/defer-and-throw/rust/core/string.rs
use crate::core_iterator::*;
use crate::core_list::*;
use crate::unions::*;

pub fn iter__3(str: String) -> StrYield {
    return StrYield { text: str, at: 0 };
}

#[derive(Clone, Debug)]
pub struct StrYield {
    pub text: String,
    pub at: i32,
}

pub fn next__5(pass: &mut StrYield) -> Union2<char, Finished> {
    let mut chr = pass.text.chars().nth((pass.at) as usize);
    if chr.is_none() {
        return Union2::<char, Finished>::U2(finished());
    }
    pass.at = pass.at + 1;
    return Union2::<char, Finished>::U1(emitted(chr.unwrap()));
}

File: examples/collections/kotlin/core/string.kt
package salvo.core.string

import salvo.*
import salvo.core.iterator.*
import salvo.core.list.*

fun iter__7(str: String): StrYield {
    return StrYield(text = str, at = 0)
}

data class StrYield(
    var text: String,
    var at: Int,
)

fun next__5(p: StrYield): Union2<Char, Finished> {
    val chr = p.text.getOrNull(p.at)
    if (chr == null) {
        return U2_2<Char, Finished>(finished())
    }
    p.at = p.at + 1
    return U2_1<Char, Finished>(emitted(chr))
}
